### PR TITLE
MUIC-441

### PR DESCRIPTION
**Pull request by mustafa-ogun-ozturk**

### CODE REVIEW

File: widgetssdk/src/main/java/com/glia/widgets/chat/ChatView.java [hunk 1]
```diff
@@ -1180,10 +1179,22 @@ private void dismissAlertDialog() {
...
-    private static Uri chooseUriByRequestCode(int requestCode, Uri galeryImgUri, Uri cameraImgUri) {
-        if (requestCode == OPEN_DOCUMENT_ACTION_REQUEST) return galeryImgUri;
-        else if (requestCode == CAPTURE_IMAGE_ACTION_REQUEST) return cameraImgUri;
-        else return null;
+    private Uri chooseUriByRequestCode(int requestCode, Uri galleryImgUri, Uri cameraImgUri) {
+        if (requestCode == OPEN_DOCUMENT_ACTION_REQUEST) return galleryImgUri;
+        else if (requestCode == CAPTURE_IMAGE_ACTION_REQUEST) {
+            Bitmap capturedImage = null;
+            try {
+                capturedImage = FileHelper.handleSamplingAndRotationBitmap(getContext(), cameraImgUri);
```
Comment (deniss-mc) [inline widgetssdk/src/main/java/com/glia/widgets/chat/ChatView.java:1187]: Create separate method for that - don't handle it here

File: widgetssdk/src/main/java/com/glia/widgets/chat/helper/FileHelper.java [hunk 2]
```diff
@@ -28,18 +34,93 @@
...
+        matrix.postRotate(degree);
+        Bitmap rotatedImg = Bitmap.createBitmap(img, 0, 0, img.getWidth(), img.getHeight(), matrix, true);
+        img.recycle();
+        return rotatedImg;
+    }
+
+    public static Uri getImageUri(Context inContext, Bitmap inImage) {
+        ByteArrayOutputStream bytes = new ByteArrayOutputStream();
+        inImage.compress(Bitmap.CompressFormat.JPEG, 100, bytes);
+        String path = MediaStore.Images.Media.insertImage(inContext.getContentResolver(), inImage, "Title", null);
```
Comment (deniss-mc) [inline widgetssdk/src/main/java/com/glia/widgets/chat/helper/FileHelper.java:122]: this looks weird

**Review by deniss-mc — CHANGES_REQUESTED** [review]

File: widgetssdk/src/main/java/com/glia/widgets/chat/helper/FileHelper.java (hunk 2 above)
Comment (mustafa-ogun-ozturk) [reply widgetssdk/src/main/java/com/glia/widgets/chat/helper/FileHelper.java:122]: In what way?

Comment (mustafa-ogun-ozturk) [reply widgetssdk/src/main/java/com/glia/widgets/chat/helper/FileHelper.java:122]: I have the image captured by the camera as URI, I have to transform it to Bitmap for the rotation process and then transform it back to URI again. I do not know any other way than the above for doing this.

**Pull request closed without merging**